lib/linux_kernel/processor.c

```diff
@@ -591,23 +564,67 @@ static int processor_x86cache_create(void)
 	return 0;
 }
 
-/* ---- Processor Interfaces ---- */
+M0_INTERNAL int m0_processor_describe(m0_processor_nr_t id,
+				      struct m0_processor_descr *pd)
+{
+	M0_PRE(pd != NULL);
+	if (id >= nr_cpu_ids)
+		return M0_ERR(-EINVAL);
+
+	return processor_x86_info_get(id, pd);
+}
+#endif
+
+#if defined CONFIG_X86_64 || defined CONFIG_AARCH64
+/**
+   Convert bitmap from one format to another. Copy cpumask bitmap to m0_bitmap.
+
+   @param dest -> Processors bitmap for Motr programs.
+   @param src -> Processors bitmap used by Linux kernel.
+   @param bmpsz -> Size of cpumask bitmap (src)
+
+   @pre Assumes memory is alloacted for outbmp and it's initialized.
+
+   @see lib/processor.h
+   @see lib/bitmap.h
+ */
+static void processors_bitmap_copy(struct m0_bitmap *dest,
+				   const cpumask_t *src,
+				   uint32_t bmpsz)
+{
+	uint32_t bit;
+	bool     val;
+
+	M0_PRE(dest->b_nr >= bmpsz);
+
+	for (bit = 0; bit < bmpsz; ++bit) {
+		val = cpumask_test_cpu(bit, src);
+		m0_bitmap_set(dest, bit, val);
+	}
+}
 
 M0_INTERNAL int m0_processors_init()
 {
+#ifdef CONFIG_X86_64
 	int rc;
 
 	M0_PRE(!processor_init);
 	rc = processor_x86cache_create();
 	processor_init = (rc == 0);
 	return M0_RC(rc);
+#elif defined CONFIG_AARCH64
+	return 0;
+#endif
 }
 
 M0_INTERNAL void m0_processors_fini()
 {
+#ifdef CONFIG_X86_64
 	M0_PRE(processor_init);
 	processor_x86cache_destroy();
 	processor_init = false;
+#elif defined CONFIG_AARCH64
+#endif
```

Comment:
Again, better make it like this:

```c
#if defined CONFIG_AARCH64
#else /* X86_64 */
	M0_PRE(processor_init);
	processor_x86cache_destroy();
	processor_init = false;
#endif
```

This way it won't be missed later when porting to some another platform.

Reply:
we need explicit checking for  both platform. as I had explained, if the  code is compiled in other  platform (other that  x86 or aarch64) . It would not  go to compilation  phase at all , since build chain will trap it in between the process. The code is ok, there is no need to worry for silent compilation

Reply:
Why, it would compile fine - and that's the problem. But looks like you've already fixed it, so I'm resolving this.